### ISSUE TITLE
[5] Search

### ISSUE THREAD

**Issue by orangewolf** [scientist-softserv/louisville-hyku#90]
# Summary

Implement search functionality per client's requests #87

# Acceptance Criteria

- [x] Only parents should be returned when a user searches the catalog for child metadata and (manual) OCR
- [x] When a user clicks on the search results, it passes the search term as param into work (via url)
- [x] If a search param exists, pre-populate the UV search with the term

# Testing Instructions

* TBD
* Suggesting that we mark this ticket as on hold until #90 and #100 is done, so that QA can test the full implementation

## Engineering Notes

- [ ] As a first pass, consider adding filter query to [catalog_controller's config.default_solr_params](https://gitlab.com/notch8/louisville-hyku/-/blob/main/app/controllers/catalog_controller.rb#L55-64)
- [ ] Will need to check various search builders to determine where to put the filter query.
- [ ] Add parent boolean to work type - consider setting boolean in create relationships job or set parent thumbnail job, (or elsewhere) add to [app_indexer](https://gitlab.com/notch8/louisville-hyku/-/blob/main/app/indexers/app_indexer.rb#L13-18)
- [ ] Pre-populating the UV search is done on a different project (not NNP). Ask the team. 
   - yale: https://github.com/yalelibrary/yul-dc-blacklight/pull/519/files
   - yale: [Pass fulltext search query to UV when loading show page](https://github.com/yalelibrary/yul-dc-blacklight/pull/559)

## resources

- solr query params - https://solr.apache.org/guide/6_6/the-standard-query-parser.html
- solr suffix cheatsheet - https://github.com/projectblacklight/blacklight-access_controls/blob/master/solr_conf/conf/schema.xml

**Comment by orangewolf** [comment]
Linked Items
---
[Bug: broken OCR search](https://gitlab.com/notch8/louisville-hyku/-/issues/90)

[Search: Parent/Child Results](https://gitlab.com/notch8/louisville-hyku/-/issues/100)

[[3] Facets don't show all imported records](https://gitlab.com/notch8/louisville-hyku/-/issues/123)

**Comment by orangewolf** [comment]
Related Merge Requests
---
[i97 - search returns parents only](https://gitlab.com/notch8/louisville-hyku/-/merge_requests/80)
!80

[i97-parents-rake-task](https://gitlab.com/notch8/louisville-hyku/-/merge_requests/89)

**Comment by orangewolf** [comment]
Shana March 2022
---
how will parent boolean be set w manual creations?

**Comment by orangewolf** [comment]
Alisha April 2022
---
this was asked in a client meeting a few weeks ago. it is not a concern of the client as they will hardly create works manually.

**Comment by orangewolf** [comment]
Alisha April 2022
---
ref #97ref [#97 (closed)](https://github.com/notch8/louisville-hyku/-/issues/97)

# expected behavior
* [x] Only parents should be returned when a user searches the catalog for child metadata

demo: https://share.getcloudapp.com/xQuq7ewe

* [x] When a user clicks on the search results, it passes the search term as param into work (via url)

demo: https://share.getcloudapp.com/L1uRBplz

* [x] If a search param exists, pre-populate the UV search with the term

<details>
<summary> demo </summary>
- I ran a search of "stare" from the catalog search which did not return any results of course because the ocr is on the child, and in this pr only parents are being returned. so I manually went to the child work. this allowed the search query to still be a part of the params, but also to demonstrate how the search query gets passed into the iframe.
![image](/uploads/dcf76a73949f253bed7771c778d6775d/image.png)
</details>

**Comment by orangewolf** [comment]
Alisha April 2022
---
added a rake task on !89 so that existing parent works can be filtered properly when searching.

from the rails console run: `rails hyku:update:is_parent_attribute`

**Comment by orangewolf** [comment]
Shana April 2022
---
this is confirmation that I ran the rake task on staging to set existing parents' is_parent attribute.

**Comment by orangewolf** [comment]
Shana May 2022
---
@alishaevn I could've sworn I updated this ticket to clarify the HOLD :upside_down: Maybe I forgot to hit enter. Anyways, we can move it forward because I tested it myself.

Originally I put it on hold thinking about Diem. It'd be easier for her to QA the entire search implementation vs this piece of it requiring her to hunt through the inspect tools, etc.

UPDATE: Looks like I put this thought in the [testing instructions section](https://gitlab.com/notch8/louisville-hyku/-/issues/97#testing-instructions)[@alishaevn](https://github.com/alishaevn) I could've sworn I updated this ticket to clarify the HOLD 🙃 Maybe I forgot to hit enter. Anyways, we can move it forward because I tested it myself.
Originally I put it on hold thinking about Diem. It'd be easier for her to QA the entire search implementation vs this piece of it requiring her to hunt through the inspect tools, etc.
UPDATE: Looks like I put this thought in the [testing instructions section](https://gitlab.com/notch8/louisville-hyku/-/issues/97#testing-instructions)
